### PR TITLE
Flush transients whens saving plugin options

### PR DESCRIPTION
**Pull request by moraleida**
### 🎫 Ticket

[ET-1287] 

### 🗒️ Description

Flushes the transients whenever the Sandbox option is toggled on/off.

### 🎥 Artifacts 

Peek 2021-11-01 17-41

### ✔️ Checklist
- [ ] I've included a changelog entry both in readme.txt and changelog.txt files. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1287]: https://theeventscalendar.atlassian.net/browse/ET-1287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Hooks.php [hunk 1]
```diff
@@ -701,4 +703,26 @@ public function register_event_compatibility_hooks() {
...
+	public function flush_transients_when_toggling_sandbox_mode( $options ) {
+
+		if ( isset( $options[ Settings::$option_sandbox ] ) &&
+			 tec_tickets_commerce_is_sandbox_mode() !== $options[ Settings::$option_sandbox ]
+		) {
+			$signup = tribe( Base_Commerce\Gateways\PayPal\Signup::class );
+			$signup->delete_transient_data();
+		}
+
+		return $options;
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:726]: Invert the logic. to avoid indentation.

**Review by bordoni — CHANGES_REQUESTED** [review]

**Comment by rafsuntaskin** [comment]
As this is related to PayPal gateway specifically. We should move the method into: `wp-content/plugins/event-tickets/src/Tickets/Commerce/Gateways/PayPal/Settings.php` and then Hook from `wp-content/plugins/event-tickets/src/Tickets/Commerce/Gateways/PayPal/Hooks.php`.

File: src/Tickets/Commerce/Hooks.php (hunk 1 above)
Comment (moraleida) [reply src/Tickets/Commerce/Hooks.php]: done.

**Comment by moraleida** [comment]
> Maybe as this is related to PayPal gateway specifically. We should move the method into: `wp-content/plugins/event-tickets/src/Tickets/Commerce/Gateways/PayPal/Settings.php` and then Hook from `wp-content/plugins/event-tickets/src/Tickets/Commerce/Gateways/PayPal/Hooks.php`.

God thinkning. I've moved the method into `Gateway\PayPal\Hooks.php`. Not sure if we need a new method in `Settings` though because all this filter is doing is calling a method of `Signup`, so no real functionality implemented in the `Hooks` file. Let me know what you think @rafsuntaskin

**Comment by bordoni** [comment]
I would rather see it on the Signup class instead of the settings one.

**Review by bordoni — APPROVED** [review]

**Pull request merged**